design-documents/graph-ql/coverage/Cart.graphqls

```diff
@@ -11,22 +11,134 @@ type CartQueryOutput {
+    email: String
+    shipping_addresses: [ShippingCartAddress]!
+    billing_address: BillingCartAddress
+    available_payment_methods: [AvailablePaymentMethod] @doc(description: "Available payment methods")
+    selected_payment_method: SelectedPaymentMethod
+    prices: CartPrices
+    total_quantity: Float!
+    is_virtual: Boolean!
+}
+
+type AvailablePaymentMethod {
+    code: String! @doc(description: "The payment method code")
+    title: String! @doc(description: "The payment method title.")
+}
+
+type SelectedPaymentMethod {
+    code: String! @doc(description: "The payment method code")
+    title: String! @doc(description: "The payment method title.")
+    purchase_order_number: String @doc(description: "The purchase order number.")
+}
+
+interface CartItemInterface @typeResolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CartItemTypeResolver") {
     id: String!
+    quantity: Float!
+    prices: CartItemPrices
+    product: ProductInterface!
+    customizable_options: [SelectedCustomizableOption]
+}
+
+type SimpleCartItem implements CartItemInterface @doc(description: "Simple Cart Item") {
+}
 
-    line_items_count: Int!
-    items_quantity: Float!
+type VirtualCartItem implements CartItemInterface @doc(description: "Virtual Cart Item") {
+}
 
-    selected_payment_method: CheckoutPaymentMethod
-    available_payment_methods: [CheckoutPaymentMethod]!
+type ConfigurableCartItem implements CartItemInterface {
+    child_sku: String! @doc(description: "SKU of the simple product corresponding to a set of selected configurable options.")
+    configurable_options: [SelectedConfigurableOption!]
+}
 
-    customer: CheckoutCustomer
-    customer_notes: String
+type DownloadableCartItem implements CartItemInterface @doc(description: "Downloadable Cart Item") {
+    links: [DownloadableProductLinks] @deprecated(description: "Type was renamed from plural to singular, also link now has different ID type")
+    links_v2: [DownloadableProductLink] @doc(description: "An array containing information about the selected links")
+    samples: [DownloadableProductSamples] @doc(description: "DownloadableProductSamples defines characteristics of a downloadable product")
+}
 
-    gift_cards_amount_used: Money
-    applied_gift_cards: [CartGiftCard]
+type BundleCartItem implements CartItemInterface {
+    bundle_options: [SelectedBundleOption!]
+}
 
-    is_multishipping: Boolean!
-    is_virtual: Boolean!
+type GiftCardCartItem implements CartItemInterface {
+    sender_name: String!
+    recepient_name: String!
+    amount: SelectedGiftCardAmount
+    message: String
+}
+
+type SelectedConfigurableOption {
+    id_v2: ID! @doc(description: "Hash which includes option ID, option type and slected values. Can be used to move configurable product to wishlist or gift registry")
```

Comment:
Tiny nitpick on the description here: the fact that it's a hash as an implementation detail the client doesn't/shouldn't know about. IDs should be opaque values to the client

Reply:
I would prefer to keep it because this is important for the use case: read from here and use when adding items to wishlist or gift registry.
It was also supposed to serve as a hint for implementors.

Reply:
> I would prefer to keep it because this is important for the use case: read from here and use when adding items to wishlist or gift registry.

I think that can be solved with a more descriptive field name. What do you think?

> It was also supposed to serve as a hint for implementors.

I think that would make more sense as an inline comment - the description is encoded into the schema and rendered as docs in tools like GraphiQL

Reply:
Okay, will convert to inline comments. This is supposed to be an ID, what name do you suggest? I think this has been discussed previously and we have not found better name so far.

Reply:
Want to make sure I really understand this type's purpose before making a suggestion, because I still don't think I'm fully grasping it.

```graphql
type SelectedConfigurableOption {
    # Hash which includes option ID, option type and slected values. Can be used to move configurable product to wishlist or gift registry
    id_v2: ID!
    id: Int!
    option_label: String!
    value_id: Int!
    value_label: String!
}
```

Is it possible we're combining 2 logically separate types into one here? I'm having a hard time understanding how an entity could conceptually have 2 separate identifiers.

Is this type holding both the information about the option (from DB), and also the state from the user's cart? Maybe a screen-shot of what UI this is modeling would help